tests/integrationv2/test_pq_handshake.py

```diff
@@ -253,6 +268,108 @@ def test_s2nc_to_s2nd_pq_handshake(managed_process, protocol, certificate, clien
         assert_s2n_negotiation_parameters(results, expected_result)
 
 
+@pytest.mark.parametrize("s2n_policy", [Ciphers.PQ_TLS_1_3_2023_06_01], ids=get_parameter_name)
+@pytest.mark.parametrize("awslc_group", [KemGroups.SecP256r1Kyber768Draft00, KemGroups.X25519Kyber768Draft00], ids=get_parameter_name)
+def test_s2nc_to_awslc_pq_handshake(managed_process, s2n_policy, awslc_group):
+
+    if not pq_enabled():
+        pytest.skip("PQ not enabled")
+
+    if "awslc" not in get_flag(S2N_PROVIDER_VERSION):
+        pytest.skip("s2n must be compiled with awslc libcrypto in order to test PQ TLS compatibility")
+
+    port = next(available_ports)
+
+    s2n_env_vars = dict()
+    s2n_env_vars["PATH"] = os.path.abspath("../../bin")
+    s2n_env_vars["LD_LIBRARY_PATH"] = os.path.abspath("../../test-deps/awslc/lib")
+
+    awslc_env_vars = dict()
+    awslc_env_vars["PATH"] = os.path.abspath("../../test-deps/awslc/bin")
+    awslc_env_vars["LD_LIBRARY_PATH"] = os.path.abspath("../../test-deps/awslc/lib")
+
+    s2nc_client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        insecure=True,
+        cipher=s2n_policy,
+        env_overrides=s2n_env_vars,
+        protocol=Protocols.TLS13)
+
+    awslc_server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        protocol=Protocols.TLS13,
+        env_overrides=awslc_env_vars,
+        curve=Curves.from_name(awslc_group.oqs_name))
+
+    awslc_server = managed_process(BoringSSL, awslc_server_options, timeout=5)
+    s2n_client = managed_process(S2N, s2nc_client_options, timeout=5)
+    expected_result = EXPECTED_RESULTS.get((s2n_policy, awslc_group), None)
+    tests_passed = 0
+
+    for s2n_results in s2n_client.get_results():
+        assert_s2n_negotiation_parameters(s2n_results, expected_result)
+        tests_passed += 1
+
+    for awslc_results in awslc_server.get_results():
+        assert_awslc_negotiation_parameters(awslc_results, expected_result)
+        tests_passed += 1
+
+    assert tests_passed is 2
+
+
+@pytest.mark.parametrize("s2n_policy", [Ciphers.PQ_TLS_1_3_2023_06_01], ids=get_parameter_name)
+@pytest.mark.parametrize("awslc_group", [KemGroups.SecP256r1Kyber768Draft00, KemGroups.X25519Kyber768Draft00], ids=get_parameter_name)
+def test_s2nd_to_awslc_pq_handshake(managed_process, s2n_policy, awslc_group):
+
+    if not pq_enabled():
+        pytest.skip("PQ not enabled")
+
+    if "awslc" not in get_flag(S2N_PROVIDER_VERSION):
+        pytest.skip("s2n must be compiled with awslc libcrypto in order to test PQ TLS compatibility")
+
+    port = next(available_ports)
+
+    s2n_env_vars = dict()
```

Comment:
Will the manual overrides be removed in the non-draft PR?

Reply:
I'm not 100% sure if the PATH/LD_LIBRARY_PATH variables already contain these values when running in the CI Docker container. If this test passes in the CI without manually adding these environment variables, I'm happy to remove them.